src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const transformState = [state];
+
+  for (const action of actions) {
+    let stateCopy = { ...transformState[transformState.length - 1] };
+
+    switch (action.type) {
```

Comment:
Use a destruction for `action`
```suggestion
  for (const action of actions) {
  
    const { type, ........ } = action;

    switch (type) {
```